### ISSUE TITLE
`playwright/no-unused-locators` incorrectly flags variables form custom `getBy*` methods

### ISSUE THREAD

**Issue by NvanWeeghel** [mskelton/eslint-plugin-playwright#445]
The `no-unused-locators` rule treats any method call matching the `getBy*` naming pattern as returning a Playwright Locator, 
even when the method is a custom factory that returns a plain class instance. 
This causes false positive "Unused locator" errors for variables that are actively used.

Code example: 
```
class MyFormHelper {
    public static async getByLabel(parentLocator: Locator, page: Page, label: string) {
      // Returns a plain class instance, not a Locator
      return new MyFormHelper(...);
    }

    public async doSomething() { ... }
  }
```

The test uses: 
```
const helper = await MyFormHelper.getByLabel(parentLocator, page, 'My Label');
  await helper.doSomething(); // variable is clearly used
```

**Comment by mskelton** [comment]
The regex is intentionally broad to cover cases like this.

```js
const myForm = page.getByTestId('form')
const myInput = myForm.getByLabel('Name')
```

We can't verify that myForm is or isn't a locator with pure static analysis, so we default to assuming it is.

My recommendation in your case is just to rename the method to something other than getByLabel
